src/Domain.jl

```diff
@@ -1988,6 +1988,201 @@ end
 
 export jacobiany!
 
+@inline function jacobianpnsderiv!(jacp::Q,y::U,p::W,t::Z,domain::D,rxnarray::Array{Int64,2},cs::Array{Float64,1},T::Float64,V::Float64,kfs::Array{Float64,1},krevs::Array{Float64,1},Nspcs::Int64,Nrxns::Int64) where {Q3<:AbstractArray,Q<:AbstractArray,U<:AbstractArray,W,Z<:Real,D<:Union{ConstantTPDomain,ConstantTADomain}}
+    @fastmath  RTinv = 1.0/(R*T)
+    @simd for rxnind = 1:Nrxns
+        if rxnarray[2,rxnind] == 0
+            @inbounds fderiv = cs[rxnarray[1,rxnind]]
+        elseif rxnarray[3,rxnind] == 0
+            @fastmath @inbounds fderiv = cs[rxnarray[1,rxnind]]*cs[rxnarray[2,rxnind]]
+        else
+            @fastmath @inbounds fderiv = cs[rxnarray[1,rxnind]]*cs[rxnarray[2,rxnind]]*cs[rxnarray[3,rxnind]]
+        end
+
+        if rxnarray[5,rxnind] == 0
+            @fastmath @inbounds rderiv = krevs[rxnind]/kfs[rxnind]*cs[rxnarray[4,rxnind]]
+        elseif rxnarray[6,rxnind] == 0
+            @fastmath @inbounds rderiv = krevs[rxnind]/kfs[rxnind]*cs[rxnarray[4,rxnind]]*cs[rxnarray[5,rxnind]]
+        else
+            @fastmath @inbounds rderiv = krevs[rxnind]/kfs[rxnind]*cs[rxnarray[4,rxnind]]*cs[rxnarray[5,rxnind]]*cs[rxnarray[6,rxnind]]
+        end
+
+        flux = fderiv-rderiv
+        @fastmath @inbounds gderiv = rderiv*kfs[rxnind]*RTinv
+
+        deriv = zeros(Nspcs)
```

Comment:
Creating arrays especially inside a loop can be very expensive, because memory allocation takes a long time. It's probably faster to create an array outside the loop and set every value to zero each iteration. Although it would be better if we could avoid having the array at all...is there a way we can just pass the perturbations you were applying to this vector directly to the parameter jacobian?